### PR TITLE
Humanpokeballs

### PR DESCRIPTION
**Pull request by GuyonBroadway**
Adds a nasty little traitor item to the game called the Rapid Human Restraint Device (codename Human Pokeball) for medical staff and clowns. Should make medical traitors a little more fun and for 6tc you can pick a guy who's day to ruin utterly.  **Fun for most involved!**

Upon contact with the target this one use ball will instantly apply a muzzle, blindfold and straight jacket to the victim in addition to knocking them down briefly.  It can be blocked by shields in the same way shields block things and also will not work if the target has over 30 melee armour on their torso. Which means the makeshift armour you can get, and indeed the tray shield, will protect you. 

[Video of this monstrosity in action.](https://youtu.be/z44aHOUNO6c)

Shoutout to Fury McFlurry for the sprite work.
bsbnorth
bsbsouth
bsbeast
bsbwest
bsbfloor

### CODE REVIEW

**Comment by GuyonBroadway** [comment]
ignore the silly commit message, bad case of the dancing retard fingers

File: code/game/objects/items/weapons/humanball.dm
Comment (JohnGinnane) [inline code/game/objects/items/weapons/humanball.dm]: Should just be '1'

File: code/modules/clothing/outfits/standard.dm
Comment (JohnGinnane) [inline code/modules/clothing/outfits/standard.dm:556]: Missing definition?

File: html/changelogs/GuyonBroadway - RapidHumanRestraintDevice .yml
Comment (JohnGinnane) [inline html/changelogs/GuyonBroadway - RapidHumanRestraintDevice .yml]: "to to"

File: code/modules/clothing/outfits/standard.dm
Comment (JohnGinnane) [reply code/modules/clothing/outfits/standard.dm:556]: Never mind..

**Review by JohnGinnane — DISMISSED** [review]
Just some typos that need to be addressed

**Comment by Delimusca** [comment]
it doesn't look or act like a pokeball in any way, but i like the concept. :ok_hand:

**Comment by ArcLumin** [comment]
How is this a pokeball?

**Comment by GuyonBroadway** [comment]
Pokeball was just a working title, seeing as I'd damn near finished the thing by the time I came up with a not-pokeball related name I jsut left the file and branch name as-is. That being said I could've renamed the pull request... and so I did.

**Comment by ghost** [comment]
Does this handle NODROP (i.e. cursed items) properly?
Either way, I don't like the implementation of yet another insta win item. I know that you can have throw mode toggled to catch things so it would be memed pretty hard by that, but all it is gonna do is mean people will have throw intent on when walking around or else they might get snagged by this.

**Comment by Swagile** [comment]
The thing is, this is a instawin item that, from the way Guyon told me, only works once.

And for 6 TC, this would just be a specialized item you'd use to assassinate someone / capture head of staff for ID, etc.

Its either this or buying a ebow and syringe gunning with mute toxin. Same deal but you get a infinite ammo stun weapon in return that can be reused while this cannot be reused.

**Comment by ghost** [comment]
Yes, but this gags, blinds and cuffs the target. There's a reason people hate mute + prod lings.

**Comment by GuyonBroadway** [comment]
Not sure, about the no-drop thing, it may just remove it from someone but seeing as when it comes to nodrop suits there's one(two?) that people can get during a traitor round I'm not sure it's going to come up that often. Does the lings flesh space-suit have a no-drop flag? 

In any case for 6tc, one use only, AND it being restricted to medical/clowns I think it's not going to be a huge problem. If it is I can either reduce/remove the knockdown so the person on the receiving end has a chance to run away. 

It should also be blocked by whatever chance a riot shield has to block things.

**Comment by Gandalf2k15** [comment]
Can we instead of having insta wins, make more skill based weaponsnss, like utility itemss

**Comment by Swagile** [comment]
Utility items?

Do you mean more emag-esque items?

**Comment by Gandalf2k15** [comment]
Precisely! Even guns, that do cool things, like ranged hacking toools Idk

**Comment by ghost** [comment]
I think this could be an interesting niche item if it were a stun supplement. You throw it on a stunned target for an insta cuff/mute/blind. Price could be lowered to accommodate for this.

Wait for Looping/Hackole/GMs to comment on this before making changes.

**Comment by GuyonBroadway** [comment]
That could work too, and if this gets accepted as is I might just make something else that works off that idea... As it stands this IS the first real thing I ever made so I'm not expecting much besides some feedback.

**Comment by Pyko1** [comment]
HOLY FUCKING SHIT!

First off wasting about 80% of the vid for set up.
And not actually telling what it does on PR.

This is absolutely worse than a 6 TC instant kill item, as this actually lets you to to capture them alive with practically no chance of getting caught. While only chance for them is someone to find em. This will actually ruin rounds for people. There is nothing worse than shit that have 0 chance of defendign against.

Not to mention this is jsut a GG win item if you use it on cap or HoS.
It might be ok if it was just normal cuffs, But Muzzle+blindfold+straightjacket instantly with 1 click for 6 TC is jsut shit.

**Comment by Pyko1** [comment]
Also this is a lot worse than ebow or any other combo, simply because no in hand icon, unless you inspect everyone one your screen GG.

**Comment by GuyonBroadway** [comment]
I could easily add an in-hand sprite, I'd just need to get my hands on one.

**Comment by GuyonBroadway** [comment]
So doing some additional testing it does not unequip cursed items... and I like that actually, you can use a No-drop implant to make any item cursed and protect yourself from this ball (or go snag the owl hardsuit). I've got fury making me an Inhand sprite as we speak, should be nice and obvious.

**Comment by Swagile** [comment]
Rounds are already ruined by people who kill everyone then space them / behead and debrain murderbone spree's. It is REALLY easy to end someone's round as a antag and do it permanently even without a 6 TC item.

And being stunned by a ebow has only one viable counter and its something that only paranoid powergamers do and thats constantly eating anti-stun chems like Epherdine. This has the throw intent counter (from what I understand of the item) and the fact that you still have to be carried off into maint by the person who captured you (who is medical / clown). You wear an obvious sprite too (the jacket, muzzle, etc. are really obvious) so people will try to save you.

I do agree, however, that there should be a check for armor. This should only be used on people who don't get armor + helmet, Captains armor, HoS armor, hardsuits, etc. so its more of a situational tool.

**Comment by Hackole** [comment]
I don't like the idea of a 6tc instantwin.

**Comment by GuyonBroadway** [comment]
I could make it so ANY shield will completely block it, so sec can easily sue a riot shield and anyone can make themselves a tray shield for protection. 

I could also reduce the throw range so you have to get close to use it, that combined with the hopefully obvious as fuck inhand sprite would make it a little less instawin.

**Comment by Hackole** [comment]
Reduced throw range and an inhand sprite would be acceptable.

**Comment by GuyonBroadway** [comment]
That's easy, and heck if imma nerf the throw range I wanna make sure that if it's batted back at you it has EXTRA range.

Admit it, the idea of someone going in for the GG-no-re strike only to have literally backfire on them sounds funny does it not?

**Comment by Pyko1** [comment]
The thing is you are killed by most of the "insta win" shit, but this actually leaves a chance that is actually almost impossible top estimate due to being blind folded.

So you get to choose from ghosting and no ress, or waiting in case some one finds you, in god knows where, since you cant see shit.

So its almost like an instant kill item that also prevents ress even if found, if properly abused.

**Comment by Pyko1** [comment]
I mean the blindfold is just an overkill if you are straigthjacketed with muzzle hence no chance of resisting even out.

Actually do the victims even get a message who did it?
Or is it just suddenly black screen, cant talk cant do shit.

**Comment by Pyko1** [comment]
Not to even mention  gamemode restrictions, Stealth nukeops all you need is this for GG win,

**Comment by Pyko1** [comment]
Ultimately this is just shit to me, we should focus in reducing 1 shot game over crap, instead of adding more of it. Sun weapons area lready OP, not to mention crap that does not even need time to cuff.

**Comment by Swagile** [comment]
Being blindfolded or not doesn't increase or decrease res chance.

You get res'd if you aren't beheaded and someone stumbles upon your body stuffed in a locker. Or if the traitor forgets to remove your uniform and you have suit sensors on a kind medical doctor can save you.

Like I said before, a check based on armor would make this more of a situational tool than a instant GG no RE item. All of the more important people on station that, if captured, would flip the station upside down have access to armor.

On top of this de-buff, it can only be used once, AND it can only be used by clown / medical doctors. Nuke ops cannot use this unless he forget to make this traitors only and remove it from nuke ops selection.

**Comment by Swagile** [comment]
Also mcbawgs has a really good idea for a decent stun system, but it would require a huge overhaul of our current combat system.

**Comment by Pyko1** [comment]
Being blindfolded is just shit, as you cant even esitmate if you have chance to be found, or observe your surroundings, its pretty much force ghost with no chance of ress.

**Comment by Pyko1** [comment]
Honestly, you can just straightjacket+muzzle people and toss them in a cell as sec, and come back 10 seconds later to collect the muzzle+straight jacket for the next patient.

File: code/datums/uplink_item.dm
Comment (ghost) [inline code/datums/uplink_item.dm]: Very awful run on sentence here. Reword it please.
Remember, if you use commas as parenthesis then sentence should still read alright after removing the parenthesis info.

**Comment by Swagile** [comment]
I don't think you are actually reading some of what im saying.

This is medical doctor / clown exclusive. You have to roll both medical doctor / clown AND traitor to be able to do this. On top of that, it is a visible sprite, and when people get used to the item they will lynch anyone using it on others as someone dragging a fully poke ball'd player = antag. It also is a one time use item unless he didn't add that suggestion of mine.

It is part stealth, part loud item due to how obvious the sprite is. This is no different from using the sleepy pen except it already does the work for you, which makes it higher in TC cost.

**Comment by GuyonBroadway** [comment]
it IS one use only, and IS only available to medical staff and for 4tc you can get a poison kit and nab one of the syringe guns just lying about in medbay which kills them GG-no-Re nearly as fast as instant.

**Comment by Pyko1** [comment]
It is not a visible sprite tho, and imho instant gib gg no re item is more balanced than Straightjacket/blindfold/muzzle, as that jsut begs for meta shit.

Even tho some of this shit could be fixed by jsut allowing ghosted and suicided people to be "cloned", its  hardly an issue to commit suicide without the flag, and usually when only option is ghosting you are being mostlikely griefed.

**Comment by GuyonBroadway** [comment]
Ok for some reason my latest commits aren't being pushed...

**Comment by Pyko1** [comment]
Well just because there is already 1 hit ggnore shit, we should not add more of em. And there should be at least some kind of counters to poison kit as well.

**Comment by GuyonBroadway** [comment]
It's also worth noting that besides the knockdown you CAN run away while wearing a straight jacket and it does not slow you down at all.

**Comment by Pyko1** [comment]
Run away while practically blind and unable to even scram for help...

**Comment by Pyko1** [comment]
Still, should be cuffs+mute toxin at worst really. This is just begging for metagriff, as this makes more than easy to remove someone from round while still forcing them to ghost, thus ensuring there is no coming back.

**Comment by GuyonBroadway** [comment]
AHAH! I have done it!

**Comment by GuyonBroadway** [comment]
Ok we got us some sexy inhand sprites thanks to Fury, pics added to the description

**Comment by GuyonBroadway** [comment]
Alrighty added a check such that if you have 30 chest melee armour or more you are protected from the ball

**Comment by Spacedong** [comment]
>Upon contact with the target this one use ball will instantly apply a muzzle, blindfold and straight jacket to the victim.

## **NO FUN ALLOWED**

**Comment by GuyonBroadway** [comment]
Can be blocked by shields and does not work if you have more than 30 torso armour (aka all of sec and the assistant makeshift stuff)

**Comment by KayeArray** [comment]
I'm going to echo the others here and say that the game really doesn't need any more instawin items, we already have stuns for that. Instawin is a shit meta and we need to try to move away from it rather than contributing to it more.

**Comment by Swagile** [comment]
Until the current game system is changed, insta gg no re after stun is how the server will work and we have to work around that. My purposed armor checks stop most of the important classes from getting gg no re'd besides maybe CMO and RD since they don't wear their hardsuits 24/7. Either then that it becomes a situational tool and a really shitty one at that since there are better alternatives as stated before (toxins kit) that are just as bad if not worse gameplay wise and can be reused much more than this one 6TC item.

If headadmins still object even besides this, then *shrug. I am just suggesting under the current circumstances we have.

**Comment by ghost** [comment]
If you make counters, then it will force people to counter meta.
There's a good reason some people loathe random searching people, it's a counter meta to antags having gear. There's a good reason that some people loathe anti-stun implants, it's a niche counter meta to stuns which are the main tools of fighting.

If you don't add counters, then it's a no fun insta win item that removes a player. A lot like ling.

Counter meta game design is bad design.

**Comment by GuyonBroadway** [comment]
So for now I'm going to call this "finished" which is to say it works and it's now a question as to weather or not the idea itself is a good one. There are other balance changed I COULD make and I may well just if the current setup still feels a little terrible.  going to update the description to reflect how it currently works.

**Comment by ghost** [comment]
This is the most fun update that's been made since the rebase, and it's been made 100% by a hippie station player instead of being directly ripped from /tg/.

It's being shit on because fun is literally hitler, and everyone would rather add a disintegrate tome to the traitor uplink than the worlds loudest and most obvious stun which can only be used once.

**Comment by Swagile** [comment]
Where is the disintegrate tome PR?

Also, we are critcisizing it for its balance and possible gameplay issues, which I have already adressed and Guyon has taken to heart and fixed. There is nothing wrong with that, because balance is something that is very hard to achieve in a game where people can blow up half the station and the only thing stopping them is antag status.

**Comment by Swagile** [comment]
So whats the dealio on this after the proposed nerfs to it have been added?

Y/N?

**Comment by GuyonBroadway** [comment]
It needs another review and I have one other method of balancing I could use based off the persons HP if the armour one isn't good enough.

**Comment by SplergeBucket** [comment]
i cant think of anything worse than this
no no no

**Comment by GuyonBroadway** [comment]
@SplergeBucket  Care to give more detail? Can anything  be done to improve this? What about the suggested and now implemented nerfs? Can you think of any additional/alternative changes that could make the idea work?

It's OK if you think it's an utterly unworkable idea, this IS the first thing I ever did and I was balancing it around the idea that it has a fairly hight cost for a one use item AND is job restricted. I also need to make a video showing off the changes really.

**Comment by ghost** [comment]
Honestly, I don't think is going to work. Since Looping and I disagree, I'll be denying it.

**Pull request closed without merging**
